win32/tests/file_win32_ut/file_win32_ut.c

```diff
@@ -0,0 +1,1011 @@
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+static HANDLE fake_handle = (HANDLE)42;
+static PTP_POOL fake_ptp_pool = (PTP_POOL)43;
+static PTP_CLEANUP_GROUP fake_ptp_cleanup_group = (PTP_CLEANUP_GROUP)44;
+static PTP_IO fake_ptp_io = (PTP_IO)45;
+static HANDLE fake_h_event = (HANDLE)46;
+static EXECUTION_ENGINE_HANDLE fake_execution_engine = (EXECUTION_ENGINE_HANDLE)47;
+
+static void setup_file_create_expectations(const char* filename, PTP_CALLBACK_ENVIRON* captured_ptpcbe,  PTP_WIN32_IO_CALLBACK* captured_callback)
+{
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(mock_CreateFileA(filename, GENERIC_READ|GENERIC_WRITE, FILE_SHARE_READ, NULL, OPEN_ALWAYS, FILE_FLAG_OVERLAPPED|FILE_FLAG_WRITE_THROUGH, NULL));
+    STRICT_EXPECTED_CALL(mock_SetFileCompletionNotificationModes(fake_handle, FILE_SKIP_COMPLETION_PORT_ON_SUCCESS));
+    STRICT_EXPECTED_CALL(mock_InitializeThreadpoolEnvironment(IGNORED_ARG))
+        .CaptureArgumentValue_pcbe(captured_ptpcbe);
+    STRICT_EXPECTED_CALL(execution_engine_win32_get_threadpool(fake_execution_engine))
+        .CallCannotFail();
+    STRICT_EXPECTED_CALL(mock_SetThreadpoolCallbackPool(IGNORED_ARG, fake_ptp_pool))
+        .ValidateArgumentValue_pcbe(captured_ptpcbe);
+    STRICT_EXPECTED_CALL(mock_CreateThreadpoolCleanupGroup());
+    STRICT_EXPECTED_CALL(mock_SetThreadpoolCallbackCleanupGroup(IGNORED_ARG, fake_ptp_cleanup_group, IGNORED_ARG))
+        .ValidateArgumentValue_pcbe(captured_ptpcbe);
+    STRICT_EXPECTED_CALL(mock_CreateThreadpoolIo(fake_handle, IGNORED_ARG, NULL, IGNORED_ARG))
+        .ValidateArgumentValue_pcbe(captured_ptpcbe)
+        .CaptureArgumentValue_pfnio(captured_callback);
+}
+
+static FILE_HANDLE get_file_handle_and_callback(const char* filename, PTP_WIN32_IO_CALLBACK* captured_callback)
+{
+    PTP_CALLBACK_ENVIRON captured_ptpcbe;
+    setup_file_create_expectations(filename, &captured_ptpcbe, captured_callback);
+
+    FILE_HANDLE file_handle = file_create(fake_execution_engine, filename, NULL, NULL);
+
+    ASSERT_IS_NOT_NULL(file_handle);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    umock_c_reset_all_calls();
+    return file_handle;
+}
+static FILE_HANDLE get_file_handle(const char* filename)
+{
+    PTP_WIN32_IO_CALLBACK captured_callback = NULL;
+    return get_file_handle_and_callback(filename, &captured_callback);
+}
+
+
+
+static FILE_HANDLE start_file_io_async(char type, unsigned char* buffer, uint32_t size, uint64_t position, FILE_CB user_callback, void* user_context, PTP_WIN32_IO_CALLBACK* captured_callback, LPOVERLAPPED* captured_ov)
+{
+    FILE_HANDLE file_handle = get_file_handle_and_callback("test_file.txt", captured_callback);
+
+    STRICT_EXPECTED_CALL(malloc(IGNORED_ARG));
+    STRICT_EXPECTED_CALL(mock_CreateEvent(IGNORED_ARG, IGNORED_ARG, IGNORED_ARG, IGNORED_ARG));

```

Comment:
>IGNORED_ARG [](start = 55, length = 11)

I think you want the "FALSE, FALSE" here, because it's actually something MSDN says you should do with the event, good to catch the bugs early